frontend/lib/DataModels/group_endpoints.dart

```diff
@@ -0,0 +1,95 @@
+import 'package:adpv_frontend/Common/consts.dart';
+
+class Field {
+  final int id;
+  final String label;
+  final String fieldType;
+  final String? unitName;
+  late bool isBelongingToGroup;
+
+  Field(
+    this.id,
+    this.label,
+    this.fieldType,
+    this.unitName,
+    this.isBelongingToGroup,
+  );
+
+  Field.fromJson(Map json)
+      : id = json["id"],
+        label = json["label"],
+        fieldType = json["fieldType"],
+        unitName = json["unitName"],
+        isBelongingToGroup = json["isBelongingToGroup"];
+}
+
+class EndpointForGroup {
+  final int id;
+  final String label;
+  late bool isBelongingToGroup;
+  Map<String, Field> fields;
+
+  EndpointForGroup(this.id, this.label, this.isBelongingToGroup, this.fields);
+
+  void sortFields() {
+    final sortedEntries = fields.entries.toList()
+      ..sort((e1, e2) => (e1.value.label.compareTo(e2.value.label)));
+    fields = Map.fromEntries(sortedEntries);
+  }
+
+  EndpointForGroup.fromJson(Map json)
+      : id = json["id"],
+        label = json["label"],
+        // ignore: unnecessary_lambdas
+        fields = {for (var e in json["fields"]) e["label"]: Field.fromJson(e)},
+        isBelongingToGroup = json["isBelongingToGroup"] {
+    for (var element in fields.entries) {
+      if ([ignoreId, ignoreLabel, ignoreField].contains(element.key)) {
+        element.value.isBelongingToGroup = true;
+      }
+    }
```

Comment:
W sumie zdałem sobie teraz sprawę, że jednak do końca nie przemyśleliśmy kwestii ignorowanych pól. W tym momencie ignorujemy `Fieldy` patrząc na ich `label`. A co w przypadku, jeżeli ktoś na backendz'ie inaczej nazwie pola `id` i `timestamp` to wtedy nasze rozwiązanie nie będzie działać. Proponuje wprowadzić dodatkowe typy `Field'ów` na backendzie (`ID`, `TIMESTAMP`) i w taki sposób to handlować. 

Tak w sumie teraz się zacząłem zastanawiać, czemu `id` i `label` musimy wrzucać w ignore? Przecież `id` i `nazwę` zawsze dostaniesz, ponieważ `backend` zawsze zwraca Ci te dane

Reply:
{
  "data": {
    "fields": [
      {
        "label": "temperature",
        "fieldType": "FOR_CHART",
        "unit": {
          "id": 1,
          "name": "°C"
        }
      }
    ],
    "data": [
      {
        "temperature": 26
      },
      {
        "temperature": 25
      },
      {
        "temperature": 25
      },
      {
        "temperature": 25
      },
      {
        "temperature": 25
      },
      {
        "temperature": 25
      },
      {
        "temperature": 25
      },
      {
        "temperature": 25
      },
      {
        "temperature": 25
      },
      {
        "temperature": 25
      },
      {
        "temperature": 25
      },
      {
        "temperature": 25
      },
      {
        "temperature": 25
      },
      {
        "temperature": 24
      },
      {
        "temperature": 24
      },
      {
        "temperature": 24
      },
      {
        "temperature": 26
      },
      {
        "temperature": 10
      }
    ]
  },
  "error": ""
}

Reply:
jest mozliwy np taki scenariusz ^ a szczerze nie ma sensu po raz kolejny robic X zapytań żeby z X endpointów posklejać sobie wszyskto, id, label i timestamp być muszą, chociażby do głupich wykresów, nie ma sensu teraz przerabiać całego backendu. Z timestampem i innym id rób jak uważasz, z mojej strony: id, label, timestamp są obowiązkowe - chyba że zamiast pisać dalej, chcemy się bawić w upiększanie kodu i od nowa turlać się po wszystkich klasach które do tej pory mamy

Reply:
Z tego co patrzę, to przy `CompareChartsView` korzystasz z `EndpointSummary`, gdzie masz zapewnione `id` i `label` z naszej bazy danych, nie z datahub'a. IMO nie ma sensu wrzucać tych `Filed'ów` w ignore. Co najwyżej `timestamp` i tylko on